### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by edumur**
Add matplotlib as optional library.
Replace hdf5 by h5py.
Add links towards the github page or website of the different linraries.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,14 +36,15 @@ Currently this means:
...
-  * `pyopengl` for 3D graphics
-    * `pyopengl` on macOS Big Sur only works with python 3.9.1+
-  * `hdf5` for large hdf5 binary format support
-  * `colorcet` for supplemental colormaps
+  * [`scipy`](https://github.com/scipy/scipy) for image processing
+  * [`pyopengl`](https://github.com/mcfletch/pyopengl) for 3D graphics
+    * [`pyopengl`](https://github.com/mcfletch/pyopengl) on macOS Big Sur only works with python 3.9.1+
+  * [`h5py`](https://github.com/h5py/h5py) for large hdf5 binary format support
+  * [`colorcet`](https://github.com/holoviz/colorcet) for supplemental colormaps
+  * [`matplotlib`](https://github.com/matplotlib/matplotlib) for supplemental colormaps
```
Comment (ntjess) [inline README.md:47]: Is it worth also mentioning some exporters can be converted to matplotlib with this dependency?

Comment (edumur) [reply README.md:47]: I would say yes, definitively.
Is there other optional stuffs to add in the README (fr matplotlib or others)?

Comment (j9ac9k) [reply README.md:47]: > I would say yes, definitively.
> Is there other optional stuffs to add in the README (fr matplotlib or others)?

Perhaps with discussing the speedups `numba` provide (I believe right now it's limited to `ImageItem.setData()` for repeated calls).  We've talked about adding some other libraries for optional support (pandas); but we have a ways to go before we get there.

If you don't think any of the existing sections would fit to discuss the optional dependent specifics, feel free to create a separate section.

**Comment by j9ac9k** [comment]
Thanks for this @edumur !

File: README.md (hunk 1 above)
Comment (ntjess) [reply README.md:47]: Also, is PyOpenGL-accelerate worth mentioning alongside PyOpenGL? I don't use either at all; I just know is that they are often referred to together.

Comment (pijyoi) [reply README.md:47]: >Perhaps with discussing the speedups numba provide (I believe right now it's limited to ImageItem.setData() for repeated calls).

Changing the image raw data is one way to trigger rescaling, yes. The other way is by changing the levels of the colorbar. For example, in #1512, changing the levels interactively through the HistogramLUTItem will call rescale several times.

**Comment by edumur** [comment]
Hi all,

I try to take into account all the inputs ;) .

**Comment by j9ac9k** [comment]
Hi @edumur 

Thanks for adding that table, I think it's useful. 

Is there a way you can left align the contents?

image

**Comment by edumur** [comment]
Done

**Comment by j9ac9k** [comment]
Hi @edumur 

looking at your fork, it appears to show the content as center aligned, not left aligned:

https://github.com/edumur/pyqtgraph

**Comment by edumur** [comment]
I misunderstood  what you meant...
Do we want this:
image
or this
image

**Comment by ixjlyons** [comment]
The latter seems fine to me - I think just the last colon on the alignment row can be removed so the whole second column is left-aligned rather than centered. The only other thing I would suggest is using reference links so the table isn't quite so wide in plain text:

```md
| Library        | Added functionality     |
|:---------------|:------------------------|
|[`scipy`]       | blah blah blah          |

[`scipy`]: https://github.com/scipy/scipy
```

**Comment by edumur** [comment]
With this:
```python

| Library | Added functionality |   
|-|-|
| [`scipy`] | <ul><li> Image processing through [`ndimage`]</li><li> Data array filtering through [`signal`] </li><ul> |
| [`pyopengl`] | <ul><li> 3D graphics </li><li> Faster image processing </li><li>Note: on macOS Big Sur only works with python 3.9.1+</li></ul> |
| [`h5py`] | <ul><li> Export in hdf5 format </li></ul> |
| [`colorcet`] | <ul><li> Add a collection of perceptually uniform colormaps </li></ul> |
| [`matplotlib`] | <ul><li> Export of PlotItem in matplotlib figure </li><li> Add matplotlib collection of colormaps </li></ul> |
| [`cupy`] | <ul><li> CUDA-enhanced image processing </li><li> Note: On Windows, CUDA toolkit must be >= 11.1 </li></ul> |
| [`numba`] | <ul><li> Faster image processing </li></ul> |

[`scipy`]: https://github.com/scipy/scipy
[`ndimage`]: https://docs.scipy.org/doc/scipy/reference/ndimage.html
[`signal`]: https://docs.scipy.org/doc/scipy/reference/signal.html
[`pyopengl`]: https://github.com/mcfletch/pyopengl
[`h5py`]: https://github.com/h5py/h5py
[`colorcet`]: https://github.com/holoviz/colorcet
[`matplotlib`]: https://github.com/matplotlib/matplotlib
[`numba`]: https://github.com/numba/numba
[`cupy`]: https://docs.cupy.dev/en/stable/install.html
```

I got this:
image

**Comment by ixjlyons** [comment]
That looks good to me. Maybe one nitpick to keep the column delimiters (`|`) aligned, at least for the first column.

**Comment by edumur** [comment]
>< Man, it was aligned and I removed it to spare space lol.
I aligned the thing and push it.

**Review by ksunden — APPROVED** [review]

**Pull request merged**